assets/js/modules/analytics-4/datastore/audience-settings.js

```diff
@@ -0,0 +1,323 @@
+
+const baseInitialState = {
+	audienceSettings: undefined,
+};
+
+const baseActions = {
+	/**
+	 * Saves the audience settings.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param {Object} settings Optional. By default, this saves whatever there is in the store. Use this object to save additional settings.
+	 * @return {Object} Object with `response` and `error`.
+	 */
+	saveAudienceSettings: createValidatedAction(
+		( settings = {} ) => {
+			invariant(
+				isPlainObject( settings ),
+				'audience settings should be an object to save.'
+			);
+		},
+		function* ( settings = {} ) {
+			yield clearError( 'saveAudienceSettings', [] );
+
+			const registry = yield Data.commonActions.getRegistry();
+			const audienceSettings = yield Data.commonActions.await(
+				registry
+					.__experimentalResolveSelect( MODULES_ANALYTICS_4 )
+					.getAudienceSettings()
+			);
+
+			const { response, error } =
+				yield fetchSaveAudienceSettingsStore.actions.fetchSaveAudienceSettings(
+					{
+						...audienceSettings,
+						...settings,
+					}
+				);
+
+			if ( error ) {
+				yield receiveError( error, 'saveAudienceSettings', [] );
+			}
+
+			return { response, error };
+		}
+	),
+
+	/**
+	 * Sets the configured audiences.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param {Array} audienceResourceNames Configured audience resource names.
+	 * @return {Object} Redux-style action.
+	 */
+	setConfiguredAudiences( audienceResourceNames ) {
+		invariant(
+			Array.isArray( audienceResourceNames ),
+			'Configured audiences should be an array.'
+		);
+
+		return {
+			type: SET_CONFIGURED_AUDIENCES,
+			payload: { audienceResourceNames },
+		};
+	},
+
+	/**
+	 * Sets the audience segmentation widget visibility.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param {boolean} isWidgetHidden Whether or not the audience segmentation widget is hidden.
+	 * @return {Object} Redux-style action.
+	 */
+	setAudienceSegmentationWidgetHidden( isWidgetHidden ) {
+		invariant(
+			typeof isWidgetHidden === 'boolean',
+			'Audience segmentation widget visibility should be a boolean.'
+		);
+
+		return {
+			type: SET_AUDIENCE_SEGMENTATION_WIDGET_HIDDEN,
+			payload: { isWidgetHidden },
+		};
+	},
+};
+
+const baseControls = {};
+
+const baseReducer = createReducer( ( state, { type, payload } ) => {
+	switch ( type ) {
+		case SET_CONFIGURED_AUDIENCES: {
+			const { audienceResourceNames } = payload;
+
+			if ( ! state.audienceSettings ) {
+				state.audienceSettings = {};
+			}
+
+			state.audienceSettings.settings = {
+				...state.audienceSettings.settings,
+				configuredAudiences: audienceResourceNames,
+			};
+
+			break;
+		}
+
+		case SET_AUDIENCE_SEGMENTATION_WIDGET_HIDDEN: {
+			const { isWidgetHidden } = payload;
+
+			if ( ! state.audienceSettings ) {
+				state.audienceSettings = {};
+			}
+
+			state.audienceSettings.settings = {
+				...state.audienceSettings.settings,
+				isAudienceSegmentationWidgetHidden: isWidgetHidden,
+			};
+
+			break;
+		}
+
+		default: {
+			return state;
```

Comment:
Seeing from other recent implementation, this should just be a break and not return anything when using the `createReducer` to create the reducer. I'm not sure if this has any unintended side effect, but we should be consistent.

Reply:
Both are identical. However, I have changed it to be consistent.